pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -323,3 +354,61 @@ func normalizeInstanceSize(ctx *workflow.Context, currentInstanceSize string, au
 func isInstanceSizeTheSame(currentDeployment *mongodbatlas.AdvancedCluster, desiredInstanceSize string) bool {
 	return desiredInstanceSize == currentDeployment.ReplicationSpecs[0].RegionConfigs[0].ElectableSpecs.InstanceSize
 }
+
+func (r *AtlasDeploymentReconciler) ensureConnectionSecrets(ctx *workflow.Context, project *mdbv1.AtlasProject, name string, connectionStrings *mongodbatlas.ConnectionStrings, deploymentResource *mdbv1.AtlasDeployment) workflow.Result {
```

Comment:
We have a function like this for `atlasdatabaseuser_controller.go`. Maybe it makes sense to merge these two methods in one. Up to you.

Reply:
Makes sense 👌